src/library_glfw.js

```diff
@@ -641,6 +676,31 @@ var LibraryGLFW = {
       }
     },
 
+    setJoystickCallback: function(winid, cbfun) {
+      var win = GLFW.WindowFromId(winid);
+      if (!win) return;
+      win.joystickFunc = cbfun;
+    },
+
+    refreshJoystick: function(joy) {
+      // TODO: refresh all joysticks, and call from render loop?
+      var j = GLFW.active.joys[joy];
+      if (!j) return;
+
+      var gamepad = navigator.getGamepads()[j.index];
```

Comment:
Reading the documentation at http://www.glfw.org/docs/latest/group__input.html#gace54cd930dcd502e118fe4021384ce1b and http://www.glfw.org/docs/latest/input_guide.html#joystick_button, it looks like a suitable usage from user code might be to do a 

    for(int i = 0; i < numJoysticks; ++i) {
        int count;
        const unsigned char* axes = glfwGetJoystickButtons(GLFW_JOYSTICK_1+i, &count);
        for(int j = 0; j < count; ++j) {
            // process axes[j];
        }
    }

In that case, there is a possible subtle bug due to an unspecced feature of the Gamepad API regarding `navigator.getGamepads()`, see https://github.com/w3c/gamepad/issues/22 and https://github.com/w3c/gamepad/issues/8. The issue is that `navigator.getGamepads()` polls new state for all gamepads, but the above line only reads data from one of them. So if there are e.g. 3 joysticks, the above loop would poll the state of each of the three joysticks three times, and process only one of them at each loop iteration. This might potentially miss out button presses, but whether that happens in practice is somewhat theoretical. Still good to keep in mind though.